src/config/load.c

```diff
@@ -64,6 +65,29 @@ cfgLoadUpdateOption(void)
 {
     FUNCTION_LOG_VOID(logLevelTrace);
 
+    // Prevent pg and repo options from being used for the default command role when they are internal. This is a cheap and cheerful
+    // way to prevent users from setting these options but still allow them to be used for other roles.
+    if (cfgCommandRole() == cfgCmdRoleDefault)
+    {
+        if (cfgOptionTest(cfgOptPg) && cfgDefOptionInternal(cfgCommand(), cfgOptPg))
+            THROW_FMT(OptionInvalidError, "option '" CFGOPT_PG "' not valid for command '%s'", cfgCommandName(cfgCommand()));
+
+        if (cfgOptionTest(cfgOptRepo) && cfgDefOptionInternal(cfgCommand(), cfgOptRepo))
+            THROW_FMT(OptionInvalidError, "option '" CFGOPT_REPO "' not valid for command '%s'", cfgCommandName(cfgCommand()));
+    }
+
+    // Make sure repo option is set for the default command role when it is not internal and more than one repo is configured or the
+    // first configured repo is not key 1.
```

Comment:
I have pushed a commit https://github.com/cmwshang/pgbackrest/commit/267e1aa493c4785bceebf12c91d5742076f5edc3. I am hoping it will run through Travis and be ready for inclusion tomorrow.